qa/sc_evm_sync_status_forger_node_corner_case.py

```diff
@@ -0,0 +1,234 @@
+            if execute_stop and time.time() - start >= 15:
+                logging.info("Disconnect sidechain nodes")
+                disconnect_sc_nodes_bi(self.sc_nodes, 0, 1)
+                logging.info("Stopping SC node 0")
+                stop_sc_node(self.sc_nodes[0], 0)
+                break
+
+            time.sleep(1)
+
+        if not sync_status:
+            fail("SC node 1 was in sync but has not detected that it is in sync")
+
+    def run_sc_node(self, sc_node_idx):
+        logging.info("Starting SC node " + str(sc_node_idx))
+        start_sc_node(sc_node_idx, self.options.tmpdir, binary=EVM_APP_BINARY)
+        wait_for_sc_node_initialization(self.sc_nodes)
+        time.sleep(3)
+
+    def restart_sc_node(self, sc_node_idx):
+        logging.info("Stopping SC node " + str(sc_node_idx))
+        stop_sc_node(self.sc_nodes[sc_node_idx], sc_node_idx)
+        time.sleep(2)
+
+        self.run_sc_node(sc_node_idx)
+
+    def run_test(self):
+        sc_node0 = self.sc_nodes[0]
+        sc_node1 = self.sc_nodes[1]
+        self.sc_ac_setup()
+
+        # Send a FT to the SC node 2 and make it a forger node
+        logging.info("Send a FT to the SC node 2")
+        sc_node1_address = sc_node1.wallet_createPrivateKeySecp256k1()["result"]["proposition"]["address"]
+
+        forward_transfer_to_sidechain(self.sc_nodes_bootstrap_info.sidechain_id,
+                                      self.nodes[0],
+                                      sc_node1_address,
+                                      50,
+                                      self.mc_return_address)
+
+        self.block_id = generate_next_block(sc_node0, "node 0", force_switch_to_next_epoch=True)
+        self.sc_sync_all()
+
+        sc_node1_vrfPubKey = sc_node1.wallet_createVrfSecret()["result"]["proposition"]["publicKey"]
+        sc_node1_blockSignPubKey = sc_node1.wallet_createPrivateKey25519()["result"]["proposition"]["publicKey"]
+
+        result = ac_makeForgerStake(sc_node1, sc_node1_address, sc_node1_blockSignPubKey, sc_node1_vrfPubKey,
+                                    convertZenToZennies(33))
+        self.sc_sync_all()
+
+        generate_next_block(sc_node0, "node 0")
+        self.sc_sync_all()
+
+        # Checking the receipt
+        tx_id = result['result']['transactionId']
+        receipt = sc_node1.rpc_eth_getTransactionReceipt(add_0x_prefix(tx_id))
+        status = int(receipt['result']['status'], 16)
+        assert_equal(1, status, "Make forger stake with native smart contract as owner should create a failed tx")
+
+        self.block_id = generate_next_block(sc_node0, "node 0", force_switch_to_next_epoch=True)
+        self.sc_sync_all()
+
+        self.block_id = generate_next_block(sc_node0, "node 0", force_switch_to_next_epoch=True)
+        self.sc_sync_all()
+
+        # -------------------------------------------------------------------------------------
+        # Test
```

Comment:
I think this test can be simplified:

1. Node0 generate 10 blocks
2. Node 1 shut down
3. Node 1 restart and generate 1 block
4. Node0-Node1 reconnection
5. Node0 generate 1 block
6. Node1 doesn't find the parentid of this block in his list (it will only contain the newly generated block) and raise the exception

Reply:
In that workflow execution of steps 3 and 4 will ensure that the generated block will be shared and recognized by both nodes as the new tip. After which the block generated in step 5 will find correlation between those present in the `SyncStatusActor` buffer of `sc_node1`.

In the current python test implementation the `sc_node1` is stopped, restarted and then it create a couple of blocks on its own before being connected to the rest of the network, this simulate the node restart performed by the infra team.
I modified the current test flow removing the fork creation on `sc_node1` because it wasn't necessary and I also lowered the number of blocks created on `sc_node0` to 500 to 200.